sparkctl/cmd/client.go

```diff
@@ -24,9 +24,22 @@ import (
 
 	"github.com/GoogleCloudPlatform/spark-on-k8s-operator/pkg/apis/sparkoperator.k8s.io/v1beta1"
 	crdclientset "github.com/GoogleCloudPlatform/spark-on-k8s-operator/pkg/client/clientset/versioned"
+	"os"
 )
 
 func buildConfig(kubeConfig string) (*rest.Config, error) {
+
+	// Check if kubeConfig exist
```

Comment:
The right fix I think should be to change `getKubeConfigPath` in `root.go` to return an empty string when the default path does not exist. `BuildConfigFromFlags` will use `InClusterConfig` if the input path is empty.

Reply:
Right, The `getKubeConfigPath` return the default path if `KUBECONFIG` env is not present. In that case, we can check if the default path exist as well, and return "" if it doesn't exist.

I'll update the PR